### ISSUE TITLE
millipede: zero-width pulses trip assertion (Trac #1712)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1704]
In the MillipedeDOMCacheMap an assertion fails if the input pulsemap contains pulses with a width of zero.

The change that lead to this behaviour is this one:
http://code.icecube.wisc.edu/projects/icecube/changeset/125877/IceCube

I found such cases using simulation sets, e.g. 11070.
There is a single event and a minimal example provided in `/data/user/tkintscher/bugs/millipede_crash`.

The comment in the code suggests a way to handle these cases: Add the charge to the next pulse and remove the offending pulse. As illustrated by the second script in the folder, doing this manually in Python before calling Millipede resolves the issue.

I think it would be more elegant to instead modify the binning in the C++ code. Upon encountering such a pulse, the variables `bins` and `raw_bins` could be decreased by one. Ideally, one should then resize the `raw_bin_edges` and `raw_charges` arrays.

Migrated from https://code.icecube.wisc.edu/ticket/1712
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:10", 
    "description": "In the MillipedeDOMCacheMap an assertion fails if the input pulsemap contains pulses with a width of zero.\n\nThe change that lead to this behaviour is this one:\nhttp://code.icecube.wisc.edu/projects/icecube/changeset/125877/IceCube\n\nI found such cases using simulation sets, e.g. 11070.\nThere is a single event and a minimal example provided in `/data/user/tkintscher/bugs/millipede_crash`.\n\nThe comment in the code suggests a way to handle these cases: Add the charge to the next pulse and remove the offending pulse. As illustrated by the second script in the folder, doing this manually in Python before calling Millipede resolves the issue.\n\nI think it would be more elegant to instead modify the binning in the C++ code. Upon encountering such a pulse, the variables `bins` and `raw_bins` could be decreased by one. Ideally, one should then resize the `raw_bin_edges` and `raw_charges` arrays.", 
    "reporter": "thomas.kintscher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067190995086", 
    "component": "combo reconstruction", 
    "summary": "millipede: zero-width pulses trip assertion", 
    "priority": "normal", 
    "keywords": "millipede pulses", 
    "time": "2016-05-19T14:22:44", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160519T15:45:44: 
* jbraun changed status from "new" to "accepted"
* jbraun changed owner from "" to "jbraun"

**Comment by icecube-trac** [comment]
Trac update at 20160520T14:04:59: jbraun changed milestone from "" to "July 2016"

**Comment by icecube-trac** [comment]
Trac update at 20160602T21:50:35: 
* jbraun changed status from "accepted" to "closed"
* jbraun commented:

> In changeset 146738:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="146738"
> Fix bin counts when removing zero-width pulses.  Closes #1704.
> ```


* jbraun changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:10: 
* olivas commented:

> Milestone July 2016 deleted


* olivas changed milestone from "July 2016" to ""
